<launch>

  <arg name="pos_pub_rate" default="100" />
  <arg name="base_frame" default ="base" />
  <arg name="final_frame" default ="right_gripper" />
  <arg name="jit_init" default ="25" />

  <arg name="joint_names" default ="[right_s0, right_s1, right_e1, right_w1]" />
  <arg name="joint_init" default ="[0, 0, 0, 0]" />

  <node name="sensor_node" pkg="bwrobot" type="sensing.py">
  	<param name="tf_ee_pub_rate" value="$(arg pos_pub_rate)" />
  	<param name="tf_base_frame" value="$(arg base_frame)" />
  	<param name="tf_final_frame" value="$(arg final_frame)" />
  </node>

  <node name="init_node" pkg="bwrobot" type="init_robot.py">
	<param name="njit" value="$(arg jit_init)" />
  	<rosparam param ="YStart" subst_value="True">$(arg joint_init)</rosparam>
  	<rosparam param ="joints" subst_value="True">$(arg joint_names)</rosparam>
  </node>


  <node name="jitter_node" pkg="bwrobot" type="jitter.py">
  	<rosparam param ="joints" subst_value="True">$(arg joint_names)</rosparam>
  </node>

</launch>
